samples/web/Continote/public/scripts/my-notes.js

```diff
@@ -0,0 +1,436 @@
+/**
+ * Copyright (c) 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This script is loaded and running whenever the my-notes.html page is open.
+ */
+(function() {
+  'use strict';
+
+  /**
+   * AuthHelper instance to simplify Firebase Auth usage.
+   *
+   * @type {?AuthHelper}
+   */
+  var authHelper_ = null;
+
+  /**
+   * Firebase Realtime Database reference for the current user's Notes.
+   *
+   * @type {?firebase.database.Reference}
+   */
+  var notesRef_ = null;
+
+  /**
+   * Various UI elements which will be manipulated through the lifecycle of
+   * this page. These are organized as part of an object for clarity and a
+   * cleaner namespace.
+   *
+   * @type {!Object}
+   * @const
+   */
+  var pageUi_ = {
+    noteList: null,
+    writeNewNoteButton: null
+  };
+
+  /**
+   * Element template used to add a Note to the Note list.
+   *
+   * @type {!string}
+   * @const
+   */
+  var noteListItemTemplate_ =
+      '<li class="mdl-list__item mdl-list__item--two-line note-list-item">' +
```

Comment:
I think it would definitely be better overall, but adds too much complexity for the scope of this sample. But, I definitely agree things could be better with more frameworks!